### PR TITLE
feat: Annotate the generated models with collection name

### PR DESCRIPTION
**Pull request by JigarJoshi**
From schema name to singular type name, there is loss of the actual collection name. To retrieve the collection name again added annotation in generated model.

### CODE REVIEW

File: code-generator/core/src/main/java/com/tigrisdata/tools/schema/TigrisDBObjectRule.java [hunk 1]
```diff
@@ -275,10 +277,24 @@ private JDefinedClass createClass(String nodeName, JsonNode node, JPackage _pack
...
                   ClassType.CLASS);
         } else {
-          String collectionName = node.get("name").asText();
-          String javaTypeName =
-              Inflector.getInstance().singularize(StringUtils.capitalize(collectionName));
+          String className = StringUtils.capitalize(nodeName);
+          String collectionName = null;
+          // if it is the sub object
+
+          if (node.has("name")) {
```
Comment (himank) [inline code-generator/core/src/main/java/com/tigrisdata/tools/schema/TigrisDBObjectRule.java:284]: So if the node doesn't have a name then we should error out?

File: code-generator/core/src/main/java/com/tigrisdata/tools/schema/TigrisDBObjectRule.java [hunk 2]
```diff
@@ -275,10 +277,24 @@ private JDefinedClass createClass(String nodeName, JsonNode node, JPackage _pack
...
+            className =
+                Inflector.getInstance()
+                    .singularize(StringUtils.capitalize(node.get("name").asText()));
+            collectionName = node.get("name").asText();
+          }
+          if (className.contains("_")) {
```
Comment (himank) [inline code-generator/core/src/main/java/com/tigrisdata/tools/schema/TigrisDBObjectRule.java:290]: can you add a comment about this?

File: code-generator/core/src/main/java/com/tigrisdata/tools/schema/TigrisDBObjectRule.java (hunk 1 above)
Comment (JigarJoshi) [reply code-generator/core/src/main/java/com/tigrisdata/tools/schema/TigrisDBObjectRule.java:284]: added shallow validation. I think we will need a layer in front here to add full validation and keep in sync with server.

File: code-generator/core/src/main/java/com/tigrisdata/tools/schema/TigrisDBObjectRule.java (hunk 2 above)
Comment (JigarJoshi) [reply code-generator/core/src/main/java/com/tigrisdata/tools/schema/TigrisDBObjectRule.java:293]: added

**Review by himank — APPROVED** [review]

**Pull request merged**